internal/controller/compaction/reconciler.go

```diff
@@ -359,6 +374,144 @@ func (r *Reconciler) createCompactionJob(ctx context.Context, logger logr.Logger
 	return job, nil
 }
 
+func recordSuccessfulJobMetrics(job *batchv1.Job) {
+	metricJobsTotal.With(prometheus.Labels{
+		druidmetrics.LabelSucceeded:     druidmetrics.ValueSucceededTrue,
+		druidmetrics.LabelFailureReason: druidmetrics.ValueFailureReasonNone,
+		druidmetrics.LabelEtcdNamespace: job.Namespace,
+	}).Inc()
+	if job.Status.CompletionTime != nil {
+		metricJobDurationSeconds.With(prometheus.Labels{
+			druidmetrics.LabelSucceeded:     druidmetrics.ValueSucceededTrue,
+			druidmetrics.LabelFailureReason: druidmetrics.ValueFailureReasonNone,
+			druidmetrics.LabelEtcdNamespace: job.Namespace,
+		}).Observe(job.Status.CompletionTime.Time.Sub(job.Status.StartTime.Time).Seconds())
+	}
+}
+
+func recordFailedJobMetrics(ctx context.Context, r *Reconciler, logger logr.Logger, job *batchv1.Job, etcd *druidv1alpha1.Etcd, jobCompletionReason string) error {
```

Comment:
instead of passing reconciler as a function arg, please make this function as a method on `Reconciler`